core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java

```diff
@@ -452,6 +475,15 @@ private void run() {
             }
         }
     }
+    
+    /**
+     * Invokes a given action on all registered {@link ContractNegotiationListener}s.
+     *
+     * @param action the action to invoke.
+     */
+    private void invokeForEach(Consumer<ContractNegotiationListener> action) {
```

Comment:
this could stay in the `ContractNegotiationObservable` abstract class, but also in `Observable` directly

Reply:
Good call, I'll move the method to `Observable`, then the negotiation managers and the async transfer process manager can share it.